algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -812,18 +827,44 @@ case class HeavyHitter[K](item: K, count: Long) extends java.io.Serializable
 
 object HeavyHitter {
 
-  def ordering[K: Ordering]: Ordering[HeavyHitter[K]] = Ordering.by { hh: HeavyHitter[K] => (hh.count, hh.item) }
+  import scala.reflect._
+
+  // Note that we can't define the ordering on `count` only.  We use a set to track heavy hitters, where equality means
+  // two elements are identical.  If we were to use `count` only, we would label different items in heavy hitters as
+  // identical simply because they have the same count (and such false duplicates may happen easily in practice).  For
+  // this reason we need to add another dimension to the ordering to prevent such false duplicates.  For this second
+  // dimension we use an item's default Java hash code because we do not want to introduce an `Ordering` constraint.
+  def ordering[K: ClassTag]: Ordering[HeavyHitter[K]] = {
```

Comment:
Why is Equiv bad news?

The reason is that in the `object Equiv` there is an implicit Equiv for every type that just defaults to `a.equals(b)`. So, with that implicit hanging out, taking an `Equiv[T]` is not a guarantee of any kind of type-safe equivalence, since if the user does nothing, the default equals gets called. So, in the case of array `Equiv[Array[Int]].equiv(Array(1), Array(1)) == false`. The correct approach, is my view, would be define Equiv for all the primitive types, String and Array in the `object Equiv` and then auto generate them in case class companions (along with Ordering and Hashing).

The reason why Hashing is lame is that it is unknown to most (I just learned of it somewhat recently) and it also has almost no good default instances, so it does not buy users much: they still have to write a lot of code, they just write it a certain way.

I really wish the typeclass story in scala were better.

About pushing Ordering onto only TopK: big fan of that. Only those using the variant that needs it should pay the price. The benefit you note with dropWhile is even more of a win.